File: aon-project/src/main/java/com/code/aon/project/TaskHolder.java
package com.code.aon.project;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.Table;

import org.apache.commons.lang.ObjectUtils;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Parameter;

import com.code.aon.common.ITransferObject;
import com.code.aon.common.dao.hibernate.PojoToStringBuilder;
import com.code.aon.project.enumeration.TaskHolderType;
import com.code.aon.registry.IRegistry;
import com.code.aon.registry.Registry;

@Entity
@Table(name="task_holder")
@PrimaryKeyJoinColumn(name="registry")
public class TaskHolder implements ITransferObject, IRegistry {
	
	private static final long serialVersionUID = 2340873810468156830L;

	private Integer id;
	private Registry registry;
    private TaskHolderType type;
    private boolean active;
	
	public TaskHolder() {
		this.type = TaskHolderType.INTERNAL;
		this.active = true;
	}

	@Id
	@Column(name="registry")
	@GeneratedValue(generator="registry_id")
	@GenericGenerator(name="registry_id", strategy="foreign", parameters = {
			@Parameter(name="property", value="registry")})
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	@OneToOne(cascade={CascadeType.PERSIST, CascadeType.MERGE})
	@org.hibernate.annotations.Cascade(value = org.hibernate.annotations.CascadeType.SAVE_UPDATE)
	@PrimaryKeyJoinColumn 
	public Registry getRegistry() {
		return registry;
	}

	public void setRegistry(Registry registry) {
		this.registry = registry;
	}

	public TaskHolderType getType() {
		return type;
	}

	public void setType(TaskHolderType type) {
		this.type = type;
	}

	public boolean isActive() {
		return active;
	}

	public void setActive(boolean active) {
		this.active = active;
	}

	@Override
	public boolean equals(Object obj) {
		if (obj == null) return false;
		if (this == obj) return true;
		if (obj.getClass() != getClass()) return false;
		final TaskHolder o = (TaskHolder) obj;
		if (o.getId() == null && getId() == null) {
			return new EqualsBuilder()
				.append(this.active, o.active)
				.append(this.registry, o.registry)
				.append(this.type, o.type)
				.isEquals();
		}
		return ObjectUtils.equals(getId(), o.getId());		
	}
	
	@Override
	public int hashCode() {
		return new HashCodeBuilder()
			.append(active)
			.append(id)			
			.append(registry)			
			.append(type)
			.toHashCode();
	}

	@Override
	public String toString() {
		return new PojoToStringBuilder(this).toString();
	}

}

File: aon-solutions/src/main/webapp/models/project/TaskHolder.js
import { Registry } from '../registry/Registry.js';

export class TaskHolder extends Registry {

    registry;
    type;
    user;
    active;
    costProfile;

    constructor(taskHolder) {
        super(taskHolder);
        if(taskHolder) {
            this.registry = taskHolder.registry || taskHolder.id;
            this.type = taskHolder.type;
            this.user = taskHolder.user;
            this.active = taskHolder.active;
            this.costProfile = taskHolder.costProfile;
        } else {
            this.active = true;
        }
    }

    getRegistry() {
        return this.registry;
    }

    getType(){
        return this.type;
    }

    setType(type) {
        this.type = type;
        return this;
    }

    getUser() {
        return this.user;
    }

    setUser(user) {
        this.user = user;
        return this;
    }

    getCostProfile() {
        return this.costProfile;
    }

    setCostProfile(costProfile){ 
        this.costProfile = costProfile;
        return this;
    }

    isActive() {
        return this.active;
    }

    setActive(active){
        this.setDirty(this.active !== active);
        this.active = active;
        return this;
    }

} 